Makefile

```diff
@@ -28,27 +30,63 @@ test: fmt vet ## Run tests.
+
 run: fmt vet ## Run a controller from your host.
 	go run ./main.go
 
 generate: fmt vet
 	go generate ./...
 
 deploy: kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
-	$(KUSTOMIZE) build config | kubectl apply -f -
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/base | kubectl apply -f -
+
+deploy-kind: kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/overlays/kind | kubectl apply -f -
 
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config | kubectl delete -f -
+	$(KUSTOMIZE) build config/base | kubectl delete -f -
+
+docker-build: ## Build docker image with the manager.
+	docker build -t ${IMG} .
 
 build-reference: kustomize
-	cd config && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
-	$(KUSTOMIZE) build config -o reference/cf-k8s-api.yaml
+	cd config/base && $(KUSTOMIZE) edit set image cloudfoundry/cf-k8s-api=${IMG}
+	$(KUSTOMIZE) build config/base -o reference/cf-k8s-api.yaml
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
 	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.2.0)
 
+HNC_VERSION ?= v0.8.0
+HNC_PLATFORM=$(shell go env GOHOSTOS)_$(shell go env GOHOSTARCH)
+HNC_BIN=$(shell pwd)/bin
+export PATH := $(HNC_BIN):$(PATH)
+hnc-install:
+	mkdir -p "$(HNC_BIN)"
+	curl -L https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-$(HNC_VERSION)/kubectl-hns_$(HNC_PLATFORM) -o "$(HNC_BIN)/kubectl-hns"
+	chmod +x "$(HNC_BIN)/kubectl-hns"
+
+	kubectl label ns kube-system hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl label ns kube-public hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl label ns kube-node-lease hnc.x-k8s.io/excluded-namespace=true --overwrite
+	kubectl apply -f https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-$(HNC_VERSION)/hnc-manager.yaml
+	kubectl rollout status deployment/hnc-controller-manager -w -n hnc-system
+	# Hierarchical namespace controller is quite asynchronous. There is no
+	# guarantee that the operations below would succeed on first invocation,
+	# so retry until they do.
+	echo -n waiting for hns controller to be ready and servicing validating webhooks
+	until kubectl create namespace ping-hnc; do echo -n .; sleep 0.5; done
+	until kubectl hns create -n ping-hnc ping-hnc-child; do echo -n .; sleep 0.5; done
+	until kubectl get namespace ping-hnc-child; do echo -n .; sleep 0.5; done
+	until kubectl hns set --allowCascadingDeletion ping-hnc; do echo -n .; sleep 0.5; done
+	until kubectl delete namespace ping-hnc --wait=false; do echo -n .; sleep 0.5; done
+	echo
+
```

Comment:
Feel like to see this moved into a `./hack/install-dependency.sh` script rather than being part of the makefile. (refer to `cf-k8s-controllers` repo for inspiration)

Also updating readme with steps to install `hns` and `hns plugins` would be helpful.